### PR TITLE
RTOS Mutex Pend Wrapper

### PR DESCRIPTION
**Pull request by ray4477**
Mutex Pend Wrapper for Micrium.
Takes in a pointer to Mutex, time in ticks to wait or 0 to wait indefinitely, and an option to block. 
Returns timestamp and error in pointers ticks and err respectively.

### CODE REVIEW

**Comment by connorl309** [comment]
looks good

**Review by manthanand — APPROVED** [review]

**Comment by manthanand** [comment]
Try fixing the merge conflicts on this PR.

File: Tasks/Src/RTOS_BPS.c [hunk 1]
```diff
@@ -11,6 +11,20 @@ void RTOS_BPS_SemPost(void) {
 
 }
 
+/**
+ * @brief   Waits for Mutex, assigns timestamp and any error to err and ticks
+ * @param   *mutex - pointer to mutex
+ * @param   timeout - timeout period, if 0 will wait forever until resource available
+ * @param   options - determines if mutex available or not
```
Comment (manthanand) [inline Tasks/Src/RTOS_BPS.c:18]: Is this comment correct?

Comment (ray4477) [reply Tasks/Src/RTOS_BPS.c:18]: yes i think so

Comment (manthanand) [reply Tasks/Src/RTOS_BPS.c:18]: no it aint, look at the options parameter in the micrium function.

File: Tasks/Inc/RTOS_BPS.h [hunk 2]
```diff
@@ -44,11 +44,12 @@ void RTOS_BPS_SemPost(void);
 void RTOS_BPS_MutexCreate(BPS_OS_MUTEX *mut, char* name);
 
 /**
- * @brief   
- * @param   
- * @return  
+ * @brief   Function waits for Mutex
+ * @param   *mutex - pointer to mutex
+ * @param   timeout - timeout period, if 0 will wait forever until resource available
+ * @param   options - determines if mutex available or not
```
Comment (manthanand) [inline Tasks/Inc/RTOS_BPS.h:50]: You gotta change the comment in both files.

**Comment by ClarkPoon** [comment]
Are you planning on changing the BPS code to call your function?

**Comment by connorl309** [comment]
Yeah, the wrappers will get finalized and then I believe once we port over
to freertos or when manthan decides we will rewrite the bps code to use the
wrappers

**Pull request merged**